filter.js

```diff
@@ -241,17 +261,17 @@ function printAllChallenges() {
 function filterFunctionSearchBar() {
     const findThis = filterSearchBar.value.toLowerCase().split(" ");
     console.log(findThis)
-    testbox.innerHTML= ""
+    testbox.innerHTML = ""
     fullApiJson.forEach(challenge => {
-        if(findThis.some(test => (test != "") && (challenge.title.toLowerCase().includes(test) || challenge.description.toLowerCase().includes(test)))){
+        if (findThis.some(test => (test != "") && (challenge.title.toLowerCase().includes(test) || challenge.description.toLowerCase().includes(test)))) {
```

Comment:
If I'm reading this right, this line checks that _some_, but not _all_ of the words in your search string matches a challenge. This means that if you type "shell online", you will find any challenge that includes the word "shell" OR the word "online". At least for me, that's not what I expect as a user.

The attempt att implementing multi-token search is admirable! But I think you probably wanted to use [`.every()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every) instead of `.some()`.